utils/filecache/interfaces.go

```diff
@@ -0,0 +1,42 @@
+package filecache
+
+import (
+	"context"
+	"io"
+
+	"github.com/ARM-software/golang-utils/utils/filesystem"
+)
+
+// IFileCache provides a filesystem cache for files and directories.
+// Cache entries are identified by a key and live for a configurable TTL (time-to-live).
+// An internal garbage collector periodically evicts expired entries; the period is also configurable.
+type IFileCache interface {
+	io.Closer
+
+	// Calls the entry provider to store an entry into the cache under the provided `key`, and starts the expiration
+	// timer based on the configured TTL.
+	// Returns an error if the path doesn't exist, the key already exists in the cache, the copy process fails or
+	// the cache is closed.
+	Store(ctx context.Context, key string) error
+
+	// Evicts the cache entry identified by `key`, removing the stored cached entry immediately.
+	// Returns an error if the key does not exist in the cache, the deleting process fails or the cache is closed.
+	Evict(ctx context.Context, key string) error
+
+	// Checks whether a cache entry with the provided `key` currently exists in the cache
+	// Returns true if the entry exists, or false if it doesn't and an error if the cache is closed.
+	Has(ctx context.Context, key string) (bool, error)
+
+	// Restore copies the cached data for the provided `key` back to `restorePath` in `restoreFilesystem`.
+	// It also resets the entry's TTL, so that the most frequently accessed files remain cached.
+	// Returns an error if the key does not exist in the cache, the copying process fails or the cache is closed.
+	Restore(ctx context.Context, key string, restoreFilesystem filesystem.FS, restorePath string) error
+}
+
+// IEntryProvider defines how to store a file or a directory into the cache.
+// Implementations might copy from a local filesystem, download over HTTP, generate a file, etc.
+type IEntryProvider interface {
+	// StoreEntry stores the entry for the given `key`` and writes it to `cacheDir` using `cacheFilesystem`.
+	// Returns the absolute path to the newly stored entry, or an error if the operation fails.
+	StoreEntry(ctx context.Context, key string, cacheFilesystem filesystem.FS, cacheDir string) (string, error)
```

Comment:
could this be called FetchEntry/RetrieveEntry rather than store. It does store the entry in the cache but the main part of the operation is to get the entry from wherever it comes from